sw/airborne/modules/sensors/temp_adc.c

```diff
@@ -81,19 +83,25 @@ static struct adc_buf temp_buf3;
 #define TEMP_ADC_SYNC_SEND FALSE
 #endif
 
-float calc_ntc(int16_t raw_temp)
+/**
+ * Calculate the NTC tempreature in celcius based on the Steinhart equation
+ */
+static float calc_ntc(int16_t raw_adc, uint16_t pull_up_r, float a, float b, float c)
 {
-  float temp_c;
-  //calc for NTC
-  temp_c = log(((10240000 / raw_temp) - 10000) * 10);
-  //temp_c = 1/(0.001129148+(0.000234125*temp_c)+(0.0000000876741*temp_c*temp_c*temp_c));
-  temp_c = 1 / (0.000603985662844 + (0.000229995493730 * temp_c) + (0.000000067653027 * temp_c *
-                temp_c * temp_c));
-  temp_c = temp_c - 273.15; //convert do celcius
+#ifdef TEMP_ADC_12BIT
```

Comment:
If you have 12 bits ADC versus a 10 bits ADC.
Originally this was written for a 10 bits, so I added this define to support 12 bits as well.